bin/database.py

```diff
@@ -965,22 +965,36 @@ def upgrade_to_26_closure(job):
 
     gear_name = gear['gear']['name']
 
+    # Checks if the specific gear tag already exists for the job
+    # Redundant now that cursor only finds jobs without the tags
+    # for tag in job['tags']:
+    #     if tag == gear_name:
+    #         return True
     # Update doc
     result = config.db.jobs.update_one({'_id': job['_id']}, {'$push': {'tags': gear_name }})
     if result.modified_count == 1:
         return True
     else:
         return 'Parallel failed: update doc ' + str(job['_id']) + ' resulted modified ' + str(result.modified_count)
 
+def upgrade_to_26_test_closure(job):
+
+    for count, tag in enumerate(job['tags'],1):
+        logging.info(count)
+        if count > 2:
+            raise Exception("Job has multiple gear tags")
```

Comment:
So, you can actually have arbitrary tags. Gear name is (supposed to be) an always-present tag, but there can be others. I think our infrastructure will add `ad-hoc` and `analysis` for you depending on how the job is enqueued, and you can set any number of your own tags with `/jobs/add?tags=xyz`.